### DISCUSSION TITLE
charm: --forcerun changing head model results

### DISCUSSION THREAD

**Discussion by grodgers1** [simnibs/simnibs#173] [7.Creating head models] [answered]
Hi again,

According to the charm command help, --forcerun only changes if the m2m folder gets overwritten. But, I ran charm on the same dataset with and without forcerun and got very different results, see two charm commands and results below:
1. charm 00000018 input_data/t1_mprage_sag_iso.nii.gz input_data/t2_space_sag_p2_iso.nii.gz --forceqform --forcerun
<img width="1008" alt="Screenshot 2023-07-24 at 15 45 31" src="https://github.com/simnibs/simnibs/assets/27735862/7e709cb0-959c-4de8-b212-72a23bae5d59">

2. charm 00000018_noforcerun input_data/t1_mprage_sag_iso.nii.gz input_data/t2_space_sag_p2_iso.nii.gz --forceqform
<img width="1008" alt="Screenshot 2023-07-24 at 15 46 02" src="https://github.com/simnibs/simnibs/assets/27735862/c32015e9-2f0f-4d5f-8683-8783f4d4a38c">

To my understanding, --forcerun shouldn't have any effect on the charm result. Any ideas why I get this behaviour?

**Comment by oulap** [comment]
Yeah the --forcerun shouldn't really change anything, so that is indeed surprising.

charm reads certain settings from a file called charm.ini (which get saved into the m2m-folder as settings.ini). Are there a differences between the settings.ini files from the two runs?

**Reply by grodgers1** [reply]
Interesting, I checked and indeed the settings.ini are different:

```shell
diff m2m_00000018/settings.ini m2m_00000018_noforcerun/settings.ini 
16c16
< #If threads is 0 the code uses all available threads, otherwise specify a positive integer number
---
> # If threads is 0 the code uses all available threads, otherwise specify a positive integer number
18a19,24
> 
> # Custom intensity parameter file (default: empty)
> # Note if a file is provided it should be in the same 
> # folder as the settings.ini file
> gmm_parameter_file = ""
> 
22c28
< init_type = "mni"
---
> init_type = "atlas"
58c64
< neck_search_bounds = [-0.4, 0.1]
---
> neck_search_bounds = [-0.3, 0.1]
67c73
< bias_kernel_width = 120
---
> bias_kernel_width = 70
``` 

This would explain the difference in results. So it appears that using --forcerun automatically set the init_type to "mni" (as I understand, default is "atlas").

**Reply by oulap** [reply, marked as answer]
I'm pretty sure that shouldn't be the case. --forcerun just removes the previous run and re-runs the whole thing again. Have you edited the charm.ini file previously? init_type = "atlas" should be the default, but in some cases the "mni" option does a better job.

**Reply by grodgers1** [reply]
Ah, that seems to be the issue. For the first run, I was using one version of simnibs (4.0b1), where charm.ini seems to have init_type = "mni". My test without --forcerun was actually using a different installation (4.0.0) where charm.ini had init_type = "atlas".

I should have double-checked the charm versions before running my test.

Sorry for the confusion!
